projects/gateway2/helm/gloo-gateway/templates/gateway/proxy-deployment.yaml

```diff
@@ -89,15 +89,25 @@ spec:
         - name: http-monitoring
           containerPort: 9091
         {{- end }}
-        exec:
-          command:
-            - wget
-            - -O
-            - /dev/null
-            - 127.0.0.1:19000/ready
-          initialDelaySeconds: 3
-          periodSeconds: 10
-          failureThreshold: 3
+{{- if $gateway.readinessProbe }}
+        readinessProbe:
+{{ toYaml $gateway.readinessProbe | indent 10}}
+{{- end }}{{/*if $gateway.readinessProbe*/}}
+{{- if $gateway.livenessProbe }}
+        livenessProbe:
+{{ toYaml $gateway.livenessProbe | indent 10}}
+{{- end }}{{/*if $gateway.livenessProbe*/}}
+{{- if $gateway.gracefulShutdown }}
+{{- if $gateway.gracefulShutdown.enabled }}
+        lifecycle:
+          preStop:
+            exec:
+              command:
+              - /bin/sh
+              - -c
+              - wget --post-data "" -O /dev/null 127.0.0.1:19000/healthcheck/fail; sleep {{ $gateway.gracefulShutdown.sleepTimeSeconds | default "10" }}
```

Comment:
will this work ok if sleepTimeSeconds is nil?

Reply:
Yes since a default is set